packages/shared-ux/table_persist/src/use_table_persist.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { Criteria } from '@elastic/eui';
+import { PropertySort } from '@elastic/eui/src/services/sort/property_sort';
```

Comment:
I would rather not deep import in case the path changes.

We can create our own type for it

```ts
type PropertySort = Criteria<any>['sort'];
```

Reply:
That makes sense! I added this type in `types.ts` ([096abab](https://github.com/elastic/kibana/pull/186997/commits/096ababced9dd39a51756bebea3b67b1dc95650b))